### ISSUE TITLE
Problems with tracking of L4 Worms?

### ISSUE THREAD

**Issue by gratzle** [Zimmer-lab/barlow_track#107]
Surprisingly, none of them did much better (like the 20-30% that we see for others) than the old tracker. I’m not completely sure why, so I’m opening this issue to try to figure out what’s going on and what we might be missing.  

---

## Main differences between my datasets and the “normal” ones

1. **Behavior:** My worms are often sleeping, unlike the original dataset where they’re mostly awake.  
2. **Worm size:** L4s are smaller than young adults, and lethargus/prelethargus worms also differ in size.  
3. **Z-plane spacing:** Because of the different sizes, I’m using a different step size between z planes, which sometimes makes neurons leave the z range.  

---

### 1. Default Barlow model (ZIM2165 young adults)

- Slightly better than the old tracker overall.  
- But it sometimes collapses on certain datasets, tracks only ~20 neurons.  
- Clearly, the hyperparameters aren’t ideal for my data, so I need to train my own model.  

### 2. L4 model trained on ZIM2165 lethargus (from scratch, hyperparameter search)

- Didn’t work very well in the first trial.  
- The training data was mostly sleeping worms, and it turns out the model is sensitive to low augmentation. So that was basically the worst possible training choice (I didn’t realize it at the time).  

### 3. L4 model trained on ZIM2165 prelethargus (awake only, from scratch, hyperparameter search)

- Did better on some of the datasets that the default model couldn’t handle.  
- Still fails on other datasets.  
- Overall, not a huge improvement over the default model.  

---

## What I think might be causing this

- The barlow model seems pretty sensitive to **low augmentation** and **edge effects**.  
- Both are issues in my data:  
  1. Worms are often immobile (sleeping).  
  2. Neurons sometimes get cropped at the top or bottom of the z-stack because of the different z-spacing.  

---

## Questions / next steps

1. How can I tweak hyperparameters to better handle:  
   - Non-moving neurons  
   - Neurons that get cut off at the edges of the z-stack  

2. How was the **fungus model** trained? Did it also deal with immobile neurons?  
   - And what about the L2/L1 worms, do they also have this cropping issue?  

3. Could I just take the **existing default model** and continue training it on L4 worms?  If yes: 
  - which datasets and how many would make sense?  
  - Also, how do I move the old `train_config.yaml` architecture into the new format?  

4. Are there any weaknesses of barlow that we should keep in mind, when training models for non-conventional conditions?

5. Any recommended strategies for dealing with neurons leaving the z-range? (padding, cropping, data augmentation, etc.)

**Comment by Charles-Fieseler-Vienna** [comment]
If you want to start from pretrained, do a newer one:
/lisc/data/scratch/neurobiology/zimmer/wbfm/TrainedBarlow/baseline_zimmer/trial_0/train_config.yaml

For fine-tuning, decrease the number of epochs (probably only 10 or so)

**Comment by Charles-Fieseler-Vienna** [comment]
More important is to redo the clustering, using "label_propagation"
